projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentService.java

```diff
@@ -269,8 +274,7 @@ public void deleteDeployment(String dataJobName) {
   }
 
   private boolean deploymentExistsOrInProgress(String dataJobName) {
-    return jobImageBuilder.isBuildingJobInProgress(dataJobName)
-        || readDeployment(dataJobName).isPresent();
+    return jobImageBuilder.isBuildingJobInProgress(dataJobName) || deploymentExists(dataJobName);
```

Comment:
nit

The prior `readDeployment(dataJobName).isPresent()`, after the changes, corresponds to `readDeploymentOptional(dataJobName).isPresent()`. 
Why swapping it with newly introduced `deploymentExists()` that duplicates this logic entirely (ideally, such a method should be reusing readDeploymentOptional.isPresent btw)?

Reply:
No the deploymentExist method is lighter because it doesn't create needless objects just to check are they present.